File: Sorting/C++/insertionsort.cpp
#include<iostream>

#define ARR_SIZE 10

using namespace std;


void InsertionSort(int a[], int n){
    for(int i=0; i<n-1; i++){
        int j=i-1;
        int temp = a[i];

        while(j>=0 && temp<a[j]){
            a[j+1] = a[j];
            j = j-1;
        }

        a[j+1] = temp;
    }
}


void printArray(int a[], int n){
    for(int i=0; i<n; i++){
        cout << a[i] << "\t";
    }
    cout << endl;
}

int main(){
    int a[ARR_SIZE] = {12, 17, 2, -1, 0, 1, 15, 100, 8, 19};

    printArray(a, ARR_SIZE);
    InsertionSort(a, ARR_SIZE);
    printArray(a, ARR_SIZE);
    
}

File: Sorting/Python/insertionsort.py
def insertion_sort(arr):
    n = len(arr)
    for i in range(n):
        j = i-1
        temp = arr[i]

        while j>=0 and temp < arr[j]:
            arr[j+1] = arr[j]
            j-=1
        
        arr[j+1] = temp

        

arr = [12, 17, 2, -1, 0, 1, 15, 100, 8, 19]
print(arr)
insertion_sort(arr)
print(arr)